pkgs/build-support/emacs/generic.nix

```diff
@@ -60,8 +60,35 @@ stdenv.mkDerivation ({
 
   LIBRARY_PATH = "${lib.getLib stdenv.cc.libc}/lib";
 
+  addEmacsNativeCompPath = true;
+
+  dontPatchELF = true;
+  dontStrip = true;
+
   postInstall = ''
-    find $out/share/emacs -type f -name '*.el' -print0 | xargs -0 -n 1 -I {} -P $NIX_BUILD_CORES sh -c "emacs --batch -f batch-native-compile {} || true"
+    find $out/share/emacs -type f -name '*.el' -print0 \
+      | xargs -0 -n 1 -I {} -P $NIX_BUILD_CORES sh -c \
+          "emacs --batch --eval=\"(add-to-list 'comp-eln-load-path \\\"$out/share/emacs/native-lisp/\\\")\" -f batch-native-compile {} || true"
```

Comment:
If our `site-start.el` is loaded, we may not have to set `comp-eln-load-path` here.

Reply:
Looks like we do need this eval, as I misunderstood when setup hooks are applied.